src/shared/watchdog.c

```diff
@@ -311,12 +311,16 @@ static int watchdog_update_timeout(void) {
 }
 
 static int watchdog_open(void) {
+        static RateLimit watchdog_open_ratelimit = { 5 * USEC_PER_SEC, 1 };
         struct watchdog_info ident;
         char **try_order;
         int r;
 
         assert(watchdog_fd < 0);
 
+        if (!ratelimit_below(&watchdog_open_ratelimit))
+                return -EWOULDBLOCK;
```

Comment:
i think it would make sense to explicit reset the ratelimit right before you write that new warning message of the previous commit. and then maybe also do an explicit watchdog open request right after, so that the warning definitely is "correct" at the point in time where we write it.